.github/workflows/macM1-e2e.yaml

```diff
@@ -0,0 +1,58 @@
+name: e2e tests on Mac M1
+
+on:
+  workflow_dispatch:
+  schedule:
+  - cron: '0 9 * * 1-5'
+
+
+jobs:
+
+  e2e-tests:
+    timeout-minutes: 45
+    runs-on: arm64
+    env:
+      M1: 1
+    steps:
+      - uses: actions/checkout@v3
+        with:
+          persist-credentials: false
+          ref: main
+      - uses: actions/setup-node@v3
+        with:
+          node-version: '16.x'
+      - uses: actions/setup-go@v3
+        with:
+          go-version: '^1.18'
+      - name: Install dependencies
+        run: npm ci
+        #The next 2 steps are a workaround for an unexpected failure in launching electron before running e2e tests
+        #Such failure is addressed in issue #2915
+      - name: add tools to PATH
+        run: |
+          mkdir -p ~/.rd/bin
+          cp -R ~/actions-runner/_work/rancher-desktop/rancher-desktop/resources/darwin/bin ~/.rd/bin
+          export PATH="~/.rd/bin:$PATH"
+      - name: Run Rancher Desktop in dev
+        run: |
+          npm run dev --no-modal-dialogs &
+          sleep 300
+          ~/.rd/bin/rdctl shutdown
+          wait
+      - name: Run e2e Tests
+        continue-on-error: false
+        run: npm run test:e2e
+      - name: Failed tests
+        if: failure()
+        run: mkdir ./e2e/reports
+      - name: Upload Artifacts
+        uses: actions/upload-artifact@v3
+        if: failure()
+        with:
+          name: e2etest-artifacts
+          path: ./e2e/reports/*
+      - name: Clean all steps
+        uses: colpal/actions-clean@v1
```

Comment:
When using third-party actions excluding things we already trust (e.g. `actions/*`, `docker/*`), _always_ use the commit hash rather than a version that can change (i.e. if the repository was taken over and somebody pushes a malicious commit).